RecoTauTag/RecoTau/plugins/RecoTauJetRegionProducer.cc

```diff
@@ -193,15 +195,17 @@ void RecoTauJetRegionProducer::produce(edm::Event& evt, const edm::EventSetup& e
     nNewJets++;
   }
 
-  // Put our new jets into the event
+    // Put our new jets into the event
   edm::OrphanHandle<reco::PFJetCollection> newJetsInEvent = evt.put(std::move(newJets), "jets");
-
+  
   // Create a matching between original jets -> extra collection
-  auto matching = std::make_unique<PFJetMatchMap>(newJetsInEvent);
-  if ( nJets ) {
-    PFJetMatchMap::Filler filler(*matching);
-    filler.insert(originalJets, matchInfo.begin(), matchInfo.end());
-    filler.fill();
+  auto matching = (nJets !=0) ? std::make_unique<PFJetMatchMap>(edm::makeRefToBaseProdFrom(edm::RefToBase<reco::Jet>(jets[0]), evt), newJetsInEvent) : std::make_unique<PFJetMatchMap>();
+  for (size_t ijet = 0; ijet < nJets; ++ijet) {
+    // JAN - FIXME - this doesn't look very elegant...
```

Comment:
@mbluj @roger-wolf going through the fixmes - the code just below seems to work just fine, but I may have overseen a more elegant solution. Will remove the commented code.